Recipes: add share

o-nly opened the pull request:
Quality: wip; Description: Share is an Emacs interface for sharing media

### Brief summary of what the package does
The package has functions for abstracting over functions you would use to share media. For example, maybe a pastebin command, maybe a command to deploy a file on your nginx server.

### Direct link to the package repository
https://github.com/o-nly/emacs-share

### Your association with the package
Maintainer

### Relevant communications with the upstream package maintainer
Very much a work in progress, but already has a few useful functions.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

purcell commented:
Thanks! Some notes:
- Please remove `:branch` from the recipe. It is redundant.
- Don't put trailing `)` on their own lines; use idiomatic lisp formatting to make the code accessible to contributors. Similarly, leave an empty line between top-level definitions.
- Keep all `require` forms at the top of the code.
- Add the missing package dependency declarations for `s` and `f`
- [This](https://github.com/o-nly/emacs-share/blob/master/share.el#L326) is very mysterious

I guess the issue I'm having overall is that despite the long Commentary, it's really not clear how one would use this, and why it would be beneficial. I feel like this might be a general "framework to rule them all" without a specific killer application, and the rather all-encompassing name underlines that. For the specific cases you've implemented, a few functions that use `kill-ring` would be sufficient and more reusable.

o-nly commented:
> Thanks! Some notes:
> 
> * Please remove `:branch` from the recipe. It is redundant.
> * Don't put trailing `)` on their own lines; use idiomatic lisp >formatting to make the code accessible to contributors. Similarly, >leave an empty line between top-level definitions.

Removed :branch. Package recipe fits in one line.

> * Keep all `require` forms at the top of the code.
> * Add the missing package dependency declarations for `s` and `f`

Removed (require 's) as it's not needed. Moved (require 'f) to top.

> * [This](https://github.com/o-nly/emacs-share/blob/master/share.el#L326) is very mysterious
> 
> I guess the issue I'm having overall is that despite the long >Commentary, it's really not clear how one would use this, and why it >would be beneficial. I feel like this might be a general "framework to >rule them all" without a specific killer application, and the rather >all-encompassing name underlines that. For the specific cases you've >implemented, a few functions that use `kill-ring` would be sufficient >and more reusable.

Right, Thanks.

o-nly commented:
I should close this; since you are waiting for upstream?
